src/c++/library/grpc_client.cc

```diff
@@ -479,14 +479,18 @@ InferenceServerGrpcClient::Create(
 }
 
 Error
-InferenceServerGrpcClient::IsServerLive(bool* live, const Headers& headers)
+InferenceServerGrpcClient::IsServerLive(
+    bool* live, const Headers& headers, const int timeout_ms)
 {
   Error err;
 
   inference::ServerLiveRequest request;
   inference::ServerLiveResponse response;
   grpc::ClientContext context;
 
+  auto deadline =
```

Comment:
I would recommend creating a utilitly function "SetDeadline" which takes the `grpc::ClientContext`  and `timeout_ms` as parameters and set the deadline on the context within this function.
Call this utility function everywhere instead.